plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -929,26 +930,32 @@ func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods [
 	if len(term.TopologyKey) == 0 {
 		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
 	}
-	matchingPodExists := false
+
 	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
 		return false, false, err
 	}
+
+	podNum := int32(1)
 	for _, existingPod := range allPods {
 		match := priorityutil.PodMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector)
 		if match {
-			matchingPodExists = true
```

Comment:
@bsalamat , re affinity, yes, it's also considered in this PR (MaxPodsPerTopologyKey is 1 for affinity); my bad remember.

> Let's assume there is one existing pod that matches the selector on a node and MaxPodsPerTopologyKey is 2

`podNum > term.MaxPodsPerTopologyKey` is equal to `matchingPodExists`; according to your case (`MaxPodsPerTopologyKey is 2`), there's not enough matched pods, so it return false.

Another case I did not handle well in this PR is the first return value: if failed to `GetNodeInfo` but succeed in next cycle (false in the first cycle, true in the last cycle), we will return true. But good things is that, the first value is used by affinity :).